src/features/layer_features_panel/atoms/layerFeaturesCollectionAtom.ts

```diff
@@ -28,101 +26,52 @@ export const featuresPanelLayerId: string =
 const isLayerMustBeEnabled =
   LAYERS_REQUIRED_BY_FEATURE_PANEL.includes(featuresPanelLayerId);
 
-// TODO: update to reatom3 - in separate PR. Clean up commented values and mocks
-export const currentFeatureIdAtom = createNumberAtom(undefined, 'currentFeatureIdAtom');
-export const layerFeaturesCollectionAtom = createAtom(
-  {
-    enabledLayersAtom,
-    mountedLayersAtom,
-    focusedGeometryAtom,
-    _setState: (state: FeatureCardCfg[]) => state,
-    reset: () => {},
-    loadFeaturesForSelection: () => {},
-  },
-  (
-    { get, schedule, create, onAction, onChange },
-    state: FeatureCardCfg[] | null = null,
-  ) => {
-    onAction('_setState', (newState) => {
-      state = [...newState];
-    });
-
-    onAction('reset', () => {
-      // @ts-expect-error needs better atom type
-      currentFeatureIdAtom.set.dispatch(undefined);
-      state = null;
-    });
-
-    onChange('focusedGeometryAtom', (focusedGeometry) => {
-      // @ts-expect-error needs better atom type
-      currentFeatureIdAtom.set.dispatch(undefined);
-      state = null;
-      const enabledLayers = get('enabledLayersAtom');
-      if (isLayerMustBeEnabled && !enabledLayers.has(featuresPanelLayerId)) {
-        return;
-      }
-      if (!isGeoJSONEmpty(focusedGeometry?.geometry))
-        schedule(async (dispatch) => {
-          const response = await getFeatureCollection(
-            focusedGeometry?.geometry,
-            featuresPanelLayerId,
-          );
-          const panelData = transformFeaturesToPanelData(response);
-          dispatch(create('_setState', panelData));
-        });
-    });
-
-    onChange('mountedLayersAtom', (mountedLayers) => {
-      if (isLayerMustBeEnabled && !mountedLayers.has(featuresPanelLayerId)) {
-        // @ts-expect-error needs better atom type
-        currentFeatureIdAtom.set.dispatch(undefined);
-        state = null;
-      } else if (state === null) {
-        // layer is mounted
-        const fg = get('focusedGeometryAtom');
-        if (!isGeoJSONEmpty(fg?.geometry))
-          schedule(async (dispatch) => {
-            // @ts-expect-error needs better atom type
-            currentFeatureIdAtom.set.dispatch(undefined);
-            const response = await getFeatureCollection(
-              fg?.geometry,
-              featuresPanelLayerId,
-            );
-            const panelData = transformFeaturesToPanelData(response);
-            dispatch(create('_setState', panelData));
-          });
-      }
-    });
-
-    return state;
-  },
-  'layerFeaturesCollectionAtom',
+export const currentFeatureIdAtom = atom<number | undefined>(
```

Comment:
I personally prefer to use `null` as initial state if that state is set explicitly

Reply:
I agree here. Changed to null